C7Engine/EntryPoints/TurnHandling.cs

```diff
@@ -4,62 +4,72 @@
 
 namespace C7Engine
 {
-    using C7GameData;
-    using System;
-    public class TurnHandling
-    {
-        public static void EndTurn()
-        {
-            GameData gameData = EngineStorage.gameData;
-            Console.WriteLine("\n*** Processing turn " + gameData.turn + " ***");
-            //Barbarians.  First, generate new barbarian units.
-            foreach (Tile tile in gameData.map.barbarianCamps)
-            {
-                //7% chance of a new barbarian.  Probably should scale based on barbarian activity.
-                int result = gameData.rng.Next(100);
-                // Console.WriteLine("Random barb result = " + result);
-                if (result < 7) {
-                    MapUnit newUnit = new MapUnit();
-                    newUnit.location = tile;
-                    newUnit.owner = gameData.players[0];
-                    newUnit.unitType = gameData.unitPrototypes["Warrior"];
-                    newUnit.hitPointsRemaining = 3;
-                    newUnit.isFortified = true; //todo: hack for unit selection
+	using C7GameData;
+	using System;
+	public class TurnHandling
+	{
+		public static void EndTurn()
+		{
+			GameData gameData = EngineStorage.gameData;
 
-                    tile.unitsOnTile.Add(newUnit);
-                    gameData.mapUnits.Add(newUnit);
-                    Console.WriteLine("New barbarian added at " + tile);
-                }
-                else if (tile.NeighborsWater() && result < 10) {
-                    MapUnit newUnit = new MapUnit();
-                    newUnit.location = tile;
-                    newUnit.owner = gameData.players[0];    //todo: make this reliably point to the barbs
-                    newUnit.unitType = gameData.unitPrototypes["Galley"];
-                    newUnit.hitPointsRemaining = 3;
-                    newUnit.isFortified = true; //todo: hack for unit selection
+			foreach (Player player in gameData.players) {
```

Comment:
Two small things, one, it should probably be only human players.  But I'm also thinking maybe this should be in an appropriately named helper method.  My first thought looking at it after two days was, "why isn't this in PlayerAI.PlayTurn", before remembering it only applies to humans.

Reply:
makes sense, I'll add a helper to get human players and iterate on those instead